app/js/controllers/main.js

```diff
@@ -1,19 +1,51 @@
-/* global Controller */
+/* global Controller, Gesture */
 
 const ADDON_FILENAME = 'addon-temp.zip';
 
 export default class MainController extends Controller {
   constructor(options) {
     super(options);
-
     this.attached = false;
+    this.waitToBeOpened();
+  }
+
+  waitToBeOpened() {
+    var self = this;
```

Comment:
We should use ES6 lambdas instead of ```self``` variables, like this:
```js
Gesture.detect.then(() => {
  this.attachView();
  this.waitToBeClosed();
});
```